Bio/PDB/__init__.py

```diff
@@ -88,3 +97,67 @@
     from .SASA import ShrakeRupley
 except ImportError:
     pass
+
+_FormatToParser = {"pdb": PDBParser(), "cif": MMCIFParser()}
+
+
+def read(handle, format, id=None):
+    """Parse a file or handle using one of the available parsers to generate a structure.
+
+    Arguments:
+     - handle    - handle to the file, or the filename as a string
+     - format    - string describing the file format. can be [pdb, cif, mmtf]
+     - id        - optional id of the structure, if None header information or then base name is used
+
+    If you have the file name in a string 'filename', use:
+
+    >>> from Bio import PDB
+    >>> filename = "1A8O.pdb"
+    >>> PDB.read(filename, format="pdb")
+
+    The function also supports file handles and gzip files
+
+    >>> from Bio import PDB
+    >>> PDB.read("1A8O.pdb.gz", format="pdb")
+    """
+    # TODO *kwargs for the parser to allow QUIET or permissive mode
+
+    # Try and give helpful error messages:
+    if not isinstance(format, str):
+        raise TypeError("Need a string for the file format (lower case)")
+    if not format:
+        raise ValueError("Format required (lower case string)")
+    if format != format.lower():
+        raise ValueError("Format string '%s' should be lower case" % format)
```

Comment:
Perhaps we can be more flexible here and be case-insensitive.

Reply:
I copied this from `SeqIO.read()`. I agree that it would be more user friendly to be case insensitive and to also accept pdb.gz and cif.gz for the CIF and PDB parser.

Reply:
I would stick with "pdb" even if gzipped, rather than overloading this argument with compression information.

I'm open to having the format names treated case insensitively here and potentially in SeqIO.

Reply:
Agree with case insensitivity. In addition, I'd move format checking to its own function. Basically a `_check_format` helper that takes a string and returns a parser class (not an instance) or throws an exception.

Reply:
Agree with @peterjc  to use "pdb" also for gzipped files.